evalml/pipelines/components/transformers/preprocessing/replace_nullable_types.py

```diff
@@ -72,13 +74,9 @@ def transform(self, X, y=None):
         """
         X_t = infer_feature_types(X, ignore_nullable_types=True)
         for col in self._nullable_int_cols:
-            new_col = X_t[col].astype("float64")
-            X_t.ww.pop(col)
-            X_t.ww[col] = new_col
+            X_t.ww[col] = X_t[col].astype("float64")
```

Comment:
Having looked at the conversation [here](https://github.com/alteryx/evalml/pull/3090#discussion_r759685131), I think it would be preferable to go with what @thehomebrewnerd suggested and explicitly initialize Woodwork on the `new_col` with the logical type that you're trying to have for the column.

Setting the dtype like this means that you're relying on Woodwork's inference to not then choose to infer the column as an unexpected logical type (for example, what if woodwork checked whether all values were actually integers and then initialized it as `IntegerNullable` again). It could also end up being a bit faster to just do the type conversion within Woodwork and not also have to do type inference.

Reply:
Done!